subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt.c

```diff
@@ -432,6 +439,30 @@ img_mgmt_upload(struct mgmt_ctxt *ctxt)
 		memset(&g_img_mgmt_state.data_sha[req.data_sha.len], 0,
 			   IMG_MGMT_DATA_SHA_LEN - req.data_sha.len);
 
+#ifdef CONFIG_IMG_ENABLE_IMAGE_CHECK
+		/* Check if the existing image hash matches the hash of the underlying data */
+		if (flash_img_init_id(&ctx, g_img_mgmt_state.area_id) == 0) {
+			struct flash_img_check fic = {
+				.match = g_img_mgmt_state.data_sha,
+				.clen = g_img_mgmt_state.size,
+			};
+
+			flash_area_close(g_img_mgmt_state.area_id);
```

Comment:
You make visible to me that (maybe) something is weird with usage of flash_img_check(). This call make the confusion to me.

Reply:
It's a nuance of using `flash_img_init_id` which leaves a flash area open (which, normally, would be closed by `flash_img_buffered_write` but is not called here so needs to be closed manually. `flash_img_check` does a new flash area open and close